_posts/2016-01-29-CocoaPods-1.0.markdown

```diff
@@ -0,0 +1,140 @@
+
+  # Has it's own copy of ShowsKit + ShowWebAuth
+  target 'ShowsiOS' do
+    pod 'ShowWebAuth'
+  end
+
+  # Has it's own copy of ShowsKit + ShowTVAuth
+  target 'ShowsTV' do
+    pod 'ShowTVAuth'
+  end
+end
+```
+> note: In this example, it would be simpler to just not use the `abstract_target` at all, but it's an example, so let me off.
+
+This change enforces a much stronger connection between the CocoaPods target, this makes it impossible to make a `Podfile` like this:
+
+``` ruby
+pod 'ShowsKit'
+```
+
+Due to it having ambiguous behavior. However, this is possbile, as it declares what target to link to:
+
+``` ruby
+pod 'ShowsKit'
+target 'ShowsiOS'
+```
+
+This works because the root level is classed as an unnamed `abstract_target`.
+
+#### More Examples
+
+Let's take a look at a [really common](https://github.com/CocoaPods/pod-template/pull/144) example that is shipped as the 0.x `pod lib create`.
+
+``` ruby
+source 'https://github.com/CocoaPods/Specs.git'
+use_frameworks!
+
+target 'ORStackView_Example', :exclusive => true do
+  pod 'ORStackView', :path => '../'
+end
+
+target 'ORStackView_Tests', :exclusive => true do
+  pod 'ORStackView', :path => '../'
+
+  pod 'Quick'
+end
+```
+
+This sets up two completely independent CocoaPods targets that are exclusive to each other. Both have a copy of the library ORStackView. There is a lot going on in this `Podfile`, contrast that to the simpler 1.0 version""
+
+``` ruby
+source 'https://github.com/CocoaPods/Specs.git'
+use_frameworks!
+
+target 'ORStackView_Example' do
+  pod "ORStackView", :path => "../"
+
+  target 'ORStackView_Tests' do
+    inherit! :search_paths
+
+    pod 'Quick'
+  end
+end
+```
+
+It has an obvious hierarchy, so you can see that Pods included in `ORStackView_Example` will be included in the `ORStackView_Tests` target, the only new thing is `inherit! :search_paths` which means "don't link Pods into here, but let this target know they exist." This `Podfile` is easier to read, simpler and has less redundant information.
+
+### Off with `pod`s `:head`
+
+Adding `:head` to a `pod` nearly always worked, except when it didn't. It is documented as this:
+
+> Finally, instead of a version, you can specify the :head flag. This will use the pod’s latest version spec version, but force the download of the ‘bleeding edge’ version. Use this with caution, as the spec might not be compatible with the source material anymore.
+
+It was a tool that was really useful back when CocoaPods was small and it was hard to persuade people it was worth support any dependency manager for their projects. That time has long passed and now the flexibility that this attribute originally offered has fallen onto the side of "not worth keeping around."It would introduce extremely hard to debug issues for developers expecting it to be "use the latest version of their code repo." You can either use the `:podspec` option or point directly to the upstream repo ( via `git`, `:svn`, etc. )
```

Comment:
in british English we keep it inside the quotes, but I'll definitely add a space - thanks!